src/v/s3/client.cc

```diff
@@ -0,0 +1,415 @@
+    header.target(target);
+    header.insert(boost::beast::http::field::user_agent, agent);
+    header.insert(boost::beast::http::field::host, host);
+    header.insert(boost::beast::http::field::content_type, "text/plain");
+    header.insert(
+      boost::beast::http::field::content_length,
+      std::to_string(payload_size_bytes));
+    header.insert(aws_header_names::x_amz_content_sha256, sig);
+    auto ec = _sign.sign_header(header, sig);
+    if (ec) {
+        return ec;
+    }
+    return header;
+}
+
+result<http::client::request_header>
+request_creator::make_list_objects_v2_request(
+  const bucket_name& name,
+  std::optional<object_key> prefix,
+  std::optional<object_key> start_after,
+  std::optional<size_t> max_keys) {
+    // GET /?list-type=2&prefix=photos/2006/&delimiter=/ HTTP/1.1
+    // Host: example-bucket.s3.<Region>.amazonaws.com
+    // x-amz-date: 20160501T000433Z
+    // Authorization: authorization string
+    http::client::request_header header{};
+    auto host = fmt::format("{}.{}", name(), _ap());
+    auto target = fmt::format("/?list-type=2");
+    std::string emptysig
+      = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855";
+    std::string agent = "redpanda.io";
+    header.method(boost::beast::http::verb::get);
+    header.target(target);
+    header.insert(boost::beast::http::field::user_agent, agent);
+    header.insert(boost::beast::http::field::host, host);
+    header.insert(boost::beast::http::field::content_length, "0");
+    header.insert(aws_header_names::x_amz_content_sha256, emptysig);
+    if (prefix) {
+        header.insert(
+          aws_header_names::prefix,
+          boost::beast::string_view{(*prefix)().data(), (*prefix)().size()});
+    }
+    if (start_after) {
+        header.insert(
+          aws_header_names::start_after,
+          boost::beast::string_view{
+            (*start_after)().data(), (*start_after)().size()});
+    }
+    if (max_keys) {
+        header.insert(aws_header_names::start_after, std::to_string(*max_keys));
+    }
+    auto ec = _sign.sign_header(header, emptysig);
+    vlog(s3_log.trace, "ListObjectsV2:\n {}", header);
+    if (ec) {
+        return ec;
+    }
+    return header;
+}
+
+result<http::client::request_header>
+request_creator::make_delete_object_request(
+  bucket_name const& name, object_key const& key) {
+    http::client::request_header header{};
+    // DELETE /{object-id} HTTP/1.1
+    // Host: {bucket-name}.s3.amazonaws.com
+    // x-amz-date:{req-datetime}
+    // Authorization:{signature}
+    // x-amz-content-sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+    //
+    // NOTE: x-amz-mfa, x-amz-bypass-governance-retention are not used for now
+    auto host = fmt::format("{}.{}", name(), _ap());
+    auto target = fmt::format("/{}", key());
+    std::string emptysig
+      = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855";
+    std::string agent = "redpanda.io";
```

Comment:
this is copied around. seems like a constexpr string-view in the http::client::: namespace to me.